third_party/nvfuser/csrc/index_compute.cpp

```diff
@@ -2869,11 +2869,7 @@ std::vector<RootPredicateInfo> Index::getReferenceRootPredicates(
     // inner-most vectorized loops are not included in the loops
     // parameter. Predicates involving vectorized loops are separately
     // generated in lower_misaligned_vectorization.
-    //
-    // Second condition is simply to avoid predication on broadcasting axes as
-    // it's not required.
-    if (consumer_stop_indexing_it == consumer_stop_index_map.end() ||
-        consumer_stop_indexing_it->second->isZeroInt()) {
```

Comment:
Is it because the extent can be zero?

Reply:
Yes!

Reply:
Does this mean we end up seeing more predicates like `0 < T.size[0]`?

Reply:
Just in case, can you leave a comment and mention we can't ignore the consumer stop index even if it's zero as the extent can be zero?

Reply:
I think other than loop rotation, it will be mostly `0 < 1`? I can not think of a case where stop index is `0`.

Reply:
I wonder if we would want to specialize a fusion for the case where there's no zero-dim tensor. Assuming that's the common case, having a compilation option flag to guarantee there's no zero-dim tensor seems reasonable. It's not clear to me how much perf overhead we would have due to the conservative assumption that there can be zero-dim tensors, but if that's something we want to optimize, it seems to make sense to have two compiled kernels, one for the common case and another as backup.

Reply:
Yeah, I agree. Don't know how much effort would be to add that option. I would wait until some user request us to improve empty tensor support. :)